### ISSUE TITLE
How about using GitHub 

### ISSUE THREAD

**Issue by yusukebe** [honojs/hono#751]

**Comment by yusukebe** [comment]
Like this: https://github.com/honojs/hono/discussions/755

**Comment by fzn0x** [comment]
There's few optimizations we can do to utilize Github Discussions:
- Integrate [Giscus](https://giscus.app/) in hono.dev we can try to put Giscus on every necessary pages in very bottom of the page, e,g. Routing https://hono.dev/api/routing will have it's own Giscus. (I can work on it)
- SEO optimization on Github Discussions / Repository Issues. Hono is popular framework, I think we don't really need to maintain SEO, but for Github Discussions and Repository Issues we may need it, so every developers can found their solutions for every problem easily if it wasn't listed in hono.dev documentation.

For more: https://www.youtube.com/watch?v=BnnevOZx7g0 this video highlights the importance of marking an answer as accepted and how Snowpack able to maximize their Github Discussions.

**Comment by yusukebe** [comment]
Hi @fzn0x 

Thanks for your suggestions. Actually, I don't think we use Discussions well. However, Discussions are good at storing the information for questions and answers.

I'll check them later, but Giscus seems to be great.

I think @NicoPlyley is also interested in this topic.

**Comment by NicoPlyley** [comment]
Yes @yusukebe I've been meaning to talk with you about discussions. A lot of time answer don't get marked, or people asking questions as a general topic so there is no option for them to mark and answer. I think some moderation with discussions would be useful.

**Comment by yusukebe** [comment]
> Yes @yusukebe I've been meaning to talk with you about discussions.

Yeah. Let's talk about discussions here.

I think the first thing we have to do is "identify the problems," like "A lot of time answer don't get marked".

@fzn0x @NicoPlyley 

Are there any other problems?

**Comment by fzn0x** [comment]
> > Yes @yusukebe I've been meaning to talk with you about discussions.
> 
> Yeah. Let's talk about discussions here.
> 
> I think the first thing we have to do is "identify the problems," like "A lot of time answer don't get marked".
> 
> @fzn0x @NicoPlyley
> 
> Are there any other problems?

We may need to migrate the most useful solutions from Discord to Github Discussions, I think you already mentioned that in one of the discussions.

**Comment by NicoPlyley** [comment]
I do notice some people post in GitHub and discord. I try and answer on both, so anyone searching can still find an answer. But that does add complexity. On discord I can add a link to the discussions. But not vice versa.

Also I think discussions can be closed after answers are found. If they are closed, someone can still comment and open it back up, it is not locked. I think it makes it a lot easier to find which discussions are still looking for input/answers.

**Comment by yusukebe** [comment]
> I try and answer on both, so anyone searching can still find an answer.

@NicoPlyley It's super helpful. Thank you very much!

> We may need to migrate the most useful solutions from Discord to Github Discussions

I also think this is a good solution. They can be stocked and googled. These are very important points.

What I'm considering is that only a few people answer the question. For example, Nico will be busy. Is there any good way to increase the number of people answering questions? Or will it be solved if we migrate Discord to Discussions?

**Comment by NicoPlyley** [comment]
I can't think of a way to directly drive engagement on GitHub discussions. On Discord an auto role can be setup, where the more people help the higher role they have on the server. And it does work in some servers, but all it really does is make your username a different color. 

I think think driving engagement with Hono directly would be the best way to get more eyes on the GitHub Discussions. I do have a few ideas in that aspect I'd be happy to go into detail with.

Lastly, maybe a section on the docs something like "Need Help?", which could take you to the official discussion areas, GitHub, Discord, and if you want to incorporate Reddit.

**Comment by yusukebe** [comment]
Thanks.

> I do have a few ideas in that aspect I'd be happy to go into detail with.

I want to know them!

> Lastly, maybe a section on the docs something like "Need Help?", which could take you to the official discussion areas, GitHub, Discord, and if you want to incorporate Reddit.

Yeah. Regarding integration between Discussions and the docs, [Giscus](https://giscus.app/) which was mentioned by @fzn0x is also good!

**Comment by NicoPlyley** [comment]
Here are my ideas to help drive engagement to Hono:

- Add a section for video tutorials on the docs, allowing creators to be featured on the Hono website. This would benefit Hono as well as the video creator. It's similar to how [Drizzle](https://orm.drizzle.team/docs/overview#video-showcase) does it as well. (I also do like the idea of having a learn section).

- On Discord we can hold events. Trying out an event where anyone can register to showcase their Hono application and answer questions afterward. I don't know if we have the quantity of people on Discord yet, however, we can hold a poll to see if it is of interest to users.

- Podcasts! Now this would require you to be available. But there are some great podcasts such as [Syntax](https://syntax.fm/) (which has mentioned Hono a few times) that love interviewing developers who contribute to open-source. If you want to see if they or others are interested, I would be happy to reach out.

- I did come up with an idea for Discussions, however, it has the possibility of only driving engagement for a short period. But if we ran a contest for the most accepted answers would receive a Hono shirt. This has a lot of difficulties though when it comes to shipping logistics across the world possibly. We would also need moderation to mark answers as correct when the original poster does not respond, especially on easy questions that come straight from the docs. Maybe it does not have to be a shirt, maybe even add their name to the docs or readme as a thank you.

These are all just brainstorming ideas, I love for anyone to give feedback and build off of these. I've had some more pop into my head that I can't recall at the moment. I can continue posting here as I come up with more ideas as well.

**Comment by NicoPlyley** [comment]
>  (I also do like the idea of having a learn section).

Expanding on this idea, possibly migrating snippets to more of a blog style or creating a section for learn. This can help with SEO as well for someone searching how to use D1 with Hono, or other basics integrating with common 3rd party libraries.

**Comment by yusukebe** [comment]
Video! Podcast! I don't know who can do it (me?), but cool!

> Maybe it does not have to be a shirt, maybe even add their name to the docs or readme as a thank you.

Adding their names on the docs is a good way, and it's easy to work on.

> Expanding on this idea, possibly migrating snippets to more of a blog style or creating a section for learning. 

Ah, I also think creating a Blog and writing articles is a good opportunity to teach the best practices of Hono. That sounds good and is easy to start.

> I can continue posting here as I come up with more ideas as well.

Yeah!

**Comment by NicoPlyley** [comment]
For the video it can be anyone that creates a video for Hono on YouTube really. There are very few videos now. But it may encourage someone to create a full tutorial knowing it will be on the docs.

For podcast it would be best to be you I feel as the creator. However, it could technically be anyone with a large knowledge/experience with Hono. 

Lastly, would a blog be best to have on VitePress? I don't know if it would be easily indexable and maintainable. If not, I'd be willing to help create something or find something that will work. But then we'd need to use a subdomain like `blog.hono.dev`

**Comment by yusukebe** [comment]
> Lastly, would a blog be best to have on VitePress? I don't know if it would be easily indexable and maintainable. If not, I'd be willing to help create something or find something that will work. But then we'd need to use a subdomain like `blog.hono.dev`

I think it's better to create a Blog in `/blog` on the existing website built with VitePress. That way, there is no need to create a new repository. Easy to contribute. We may customize the design.

**Comment by NicoPlyley** [comment]
I've been attempting to make a page for it, I can keep trying but if there any vitepress experts that could give a hand. I'm not very familiar with it in regards to custom layouts or Vue.

**Comment by yusukebe** [comment]
> My biggest issue is getting the sidebar away on a blog page since it could be incredibly long.

Ah... I see.

By the way, I was referring to the Bun website, as I mentioned in https://github.com/orgs/honojs/discussions/2700

https://bun.sh/

**Comment by NicoPlyley** [comment]
Hi @yusukebe,
I was thinking we could add Discussions to the Discord as well to help gain visibility. When someone posts a new discussion it will also post it in a Discord feed with the link to the Discussion to answer it.

Let me know what you think. All we would need to do is create a webhook in GitHub

**Comment by NicoPlyley** [comment]
Great! @yusukebe I will test it out and tweak it then once it's good in the Hono GitHub settings we can have it call a webhook for discussions

**Comment by NicoPlyley** [comment]
Hi @yusukebe,
I wanted to bring up another thing I've noticed here on GitHub. There are a lot of open issues that have already been resolved, or they were a user mistake that was never closed. Occasionally, I will tag you to close, but I know you are busy, and I don't want to add even more to your notifications.

This may not be a priority or a worry to you; how it is now may be fine. However, I wanted to bring up the option of using GitHub actions to close stale issues after a long period of inactivity. I have seen other libraries do this and incorporate a label that will never auto-close since some issues or enhancements may not be touched until the next major release.

Or if you'd prefer a mention in an issue that can be closed when they are seen, I don't mind doing that either. Let me know what you think when you get a chance.

P.S.
I haven't had the chance to work on what we discussed above, but I'll let you know once I do.

**Comment by EdamAme-x** [comment]
@NicoPlyley I agree with you.

I think we can consider implementing this.
https://github.com/actions/stale

**Comment by yusukebe** [comment]
@NicoPlyley @EdamAme-x 

> However, I wanted to bring up the option of using GitHub actions to close stale issues after a long period of inactivity.

Thank you for the suggestion and for caring about me. I agree with it. I'd like to try it.
